plugins/automation/lib/discourse_automation/event_handlers.rb

```diff
@@ -75,7 +75,9 @@ def self.handle_user_updated(user, new_user: false)
         .find_each do |automation|
           once_per_user = automation.trigger_field("once_per_user")["value"]
           if once_per_user &&
-               UserCustomField.exists?(name: DiscourseAutomation::CUSTOM_FIELD, user_id: user.id)
```

Comment:
Here's one bug that will be fixed in this PR. This line should also check that the custom field's value matches the triggered automation's ID. Without doing this, it means that the `once_per_user` option in an automation rule will affect other automation rules for a given user.